src/store/presenters/presentPost.js

```diff
@@ -3,13 +3,23 @@ export default function presentPost (post, communityId) {
   const postMembership = post.postMemberships.filter(p =>
     Number(p.community) === Number(communityId)).toRefArray()[0]
   const pinned = postMembership && postMembership.pinned
+
+  let communities = post.communities.toRefArray()
+  if (post.isPublic) {
+    // TODO Update slug for public context when Public Communities and Posts context is available
+    // for now, this links to the user's All Communities view
+    communities.unshift({ name: 'Public', id: 'public', avatarUrl: '/public-icon.svg', slug: '' })
+  }
+
   return {
     ...post.ref,
     creator: post.creator,
     linkPreview: post.linkPreview,
     location: post.location,
+    isPublic: post.isPublic,
     commenters: post.commenters.toModelArray(),
     communities: post.communities.toModelArray(),
+    communitiesArray: communities,
```

Comment:
i think this could get confusing, to have both communities and communitiesArray. not sure the best solution. would it be hard to create a Community model object and add it to the modelArray instead?

Reply:
Solution - make PostCommunities into connected component and move functionality from PresentPost to there.